batch/vep/run_vep_in_variant_transforms.sh

```diff
@@ -0,0 +1,132 @@
+# through environment variables.
+#
+# The only environment variable that has to be set is (others are optional):
+#
+# VEP_CACHE: The path of the VEP cache which is a single .tar.gz file.
+#
+# The first argument is the input file (might be a VCF or a compressed VCF) and
+# the second is the output file which is always a VCF file. The watchdog file is
+# the file that will be updated by the Dataflow worker every CONSTANT time. Once
+# the file is found to be stale, the VEP process running in the VM will be 
+# killed.
+#
+# For the full list of supported environment variables and their documentation
+# check README.md.
+# Capital letter variables refer to environment variables that can be set from
+# outside. Internal variables have small letters.
+
+set -euo pipefail
+
+if [[ $# -ne 3 ]]; then
+  echo "Usage: $0 input_file output_file watchdog_file"
+  exit 1
+fi
+
+readonly species="${SPECIES:-homo_sapiens}"
+readonly assembly="${GENOME_ASSEMBLY:-GRCh38}"
+readonly fork_opt="--fork ${NUM_FORKS:-1}"
+readonly other_vep_opts="${OTHER_VEP_OPTS:---everything \
+  --check_ref --allow_non_variant}"
+readonly annotation_field_name="${VCF_INFO_FILED:-CSQ}"
+
+if [[ ! -r "${VEP_CACHE:?VEP_CACHE is not set!}" ]]; then
+  echo "ERRPR: Cannot read ${VEP_CACHE}"
+  exit 1
+fi
+
+# Check that the input file is readable.
+readonly input_file=${1}
+if [[ ! -r "${input_file}" ]]; then
+  echo "ERRPR: Cannot read ${input_file}"
+  exit 1
+fi
+
+echo "Checking the input file at $(date)"
+ls -l "${input_file}"
+
+# Make sure output file does not exist and can be written.
+readonly output_file=${2}
+if [[ -e ${output_file} ]]; then
+  echo "ERROR: ${output_file} already exist!"
+  exit 1
+fi
+mkdir -p $(dirname ${output_file})
+touch ${output_file}
+rm ${output_file}
+
+readonly vep_cache_dir="$(dirname ${VEP_CACHE})"
+readonly vep_cache_file="$(basename ${VEP_CACHE})"
+pushd ${vep_cache_dir}
+if [[ -d "${species}" ]]; then
+  echo "The cache is already decompressed; found ${species} at $(date)"
+else
+  echo "Decompressing the cache file ${vep_cache_file} started at $(date)"
+  tar xzvf "${vep_cache_file}"
+  if [[ ! -d "${species}" ]]; then
+    echo "Cannot find directory ${species} after decompressing ${vep_cache_file}!"
+    exit 1
+  fi
+fi
+popd
+
+readonly vep_command="./vep -i ${input_file} -o ${output_file} \
+  --dir ${vep_cache_dir} --offline --species ${species} --assembly ${assembly} \
+  --vcf --allele_number --vcf_info_field ${annotation_field_name} ${fork_opt} \
+  ${other_vep_opts}"
+echo "VEP command is: ${vep_command}"
+
+echo "Running vep started at $(date)"
+# The next line should not be quoted since we want word splitting to happen.
+
+readonly watchdog_file=${3}
+
+${vep_command} &
+
+pid_vep="$!"
+while ps -p ${pid_vep} > /dev/null
+do
+  next_wait_time=0
+  until start=`gsutil cat ${watchdog_file}` || [ ${next_wait_time} -eq 4 ]; do
+    sleep $(( next_wait_time++ ))
+  done
+  declare -i start
+  end=`date +%s`
+  declare -i end
```

Comment:
why is 'end' declared after being assigned?

Reply:
Updated.